scripts/analysis_1/nbconverted/2_generate_data_randVector.py

```diff
@@ -0,0 +1,270 @@
+
+NN_dir = base_dir + "/models/" + NN_architecture
+latent_dim = NN_architecture.split('_')[-1]
+
+mapping_file = os.path.join(
+    base_dir,
+    "data",
+    "metadata",
+    "sample_annotations.tsv")
+
+normalized_data_file = os.path.join(
+    base_dir,
+    "data",
+    "input",
+    "train_set_normalized.pcl")
+
+model_encoder_file = glob.glob(os.path.join(
+    NN_dir,
+    "tybalt_2layer_{}latent_encoder_model.h5".format(latent_dim)))[0]
+
+weights_encoder_file = glob.glob(os.path.join(
+    NN_dir,
+    "tybalt_2layer_{}latent_encoder_weights.h5".format(latent_dim)))[0]
+
+model_decoder_file = glob.glob(os.path.join(
+    NN_dir,
+    "tybalt_2layer_{}latent_decoder_model.h5".format(latent_dim)))[0]
+
+weights_decoder_file = glob.glob(os.path.join(
+    NN_dir,
+    "tybalt_2layer_{}latent_decoder_weights.h5".format(latent_dim)))[0]
+
+
+# In[4]:
+
+
+# Output file
+new_dir = os.path.join(local_dir, "Data", "Batch_effects", "simulated")
+
+analysis_dir = os.path.join(new_dir, analysis_name)
+
+if os.path.exists(analysis_dir):
+    print('Directory already exists: \n {}'.format(analysis_dir))
+else:
+    print('Creating new directory: \n {}'.format(analysis_dir))
+os.makedirs(analysis_dir, exist_ok=True)
+
+    
+simulated_data_file = os.path.join(
+    local_dir,
+    "Data",
+    "Batch_effects",
+    "simulated",
+    analysis_name,
+    "simulated_data.txt.xz")
+
+
+# In[5]:
+
+
+# Load saved models
+loaded_model = load_model(model_encoder_file)
+loaded_decode_model = load_model(model_decoder_file)
+
+loaded_model.load_weights(weights_encoder_file)
+loaded_decode_model.load_weights(weights_decoder_file)
+
+
+# ### Get experiment ids
+
+# In[6]:
+
+
+# Read in metadata
+metadata = pd.read_table(
+    mapping_file, 
+    header=0, 
+    sep='\t', 
+    index_col=0)
+
+metadata.head()
+
+
+# In[7]:
+
+
+map_experiment_sample = metadata[['sample_name', 'ml_data_source']]
+map_experiment_sample.head()
+
+
+# In[8]:
+
+
+experiment_ids = np.unique(np.array(map_experiment_sample.index))
+print("There are {} experiments in the compendium".format(len(experiment_ids)))
+
+
+# ### Set user selected experiment id or randomly select experiment id
+
+# In[9]:
+
+
+if not experiment_id:
+    selected_experiment_id = np.random.RandomState(randomState).choice(experiment_ids)
+else:
+    selected_experiment_id = experiment_id
+
+
+# ### Get samples belonging to selected experiment
+
+# In[10]:
+
+
+selected_metadata = metadata.loc[selected_experiment_id]
+selected_metadata.head()
+
+
+# In[11]:
+
+
+print("There are {} samples in experiment {}".format(selected_metadata.shape[0], selected_experiment_id))
+
+
+# In[12]:
+
+
+sample_ids = list(selected_metadata['ml_data_source'])
+sample_ids
+
+
+# In[13]:
+
+
+normalized_data = pd.read_table(
+    normalized_data_file,
+    header=0,
+    sep='\t',
+    index_col=0).T
+
+normalized_data.head()
+
+
+# In[14]:
+
+
+# Check that samples are in gene expression dataset otherwise exit
```

Comment:
It's not exiting so the comment is wrong.

Reply:
This was a comment that I added for my future self, but I don't think I will be doing the check this way anymore so I've changed the comment.